pkg/controller/ecs/service/setup.go

```diff
@@ -99,9 +99,14 @@ func preCreate(_ context.Context, cr *svcapitypes.Service, obj *svcsdk.CreateSer
 }
 
 func preDelete(_ context.Context, cr *svcapitypes.Service, obj *svcsdk.DeleteServiceInput) (bool, error) {
+
+	obj.SetForce(cr.Spec.ForProvider.ForceDeletion)
+
 	if aws.StringValue(cr.Spec.ForProvider.SchedulingStrategy) == svcsdk.SchedulingStrategyReplica {
-		// Force deletion of services even if they are not scaled down.
-		obj.SetForce(true)
+		if aws.Int64Value(cr.Status.AtProvider.RunningCount) > 0 || aws.Int64Value(cr.Status.AtProvider.PendingCount) > 0 {
+			// Force deletion of services even if they are not scaled down.
+			obj.SetForce(true)
+		}
```

Comment:
This change is not necessary anymore since force is now explicitly set by the user.